### PR TITLE
fix: ensure signing out does not error

### PR DESCRIPTION
**Pull request by Prithpal-Sooriya**
## Explanation

We do not need to perform state assertions when signing out. This is to ensure if users get into weird situations, they can still sign out without error.

## References

Fixes: https://github.com/MetaMask/MetaMask-planning/issues/3709

### `@metamask/profile-sync-controller`

- **CHANGED** removed assert logic in `performSignOut()` method

## Checklist

- [x] I've updated the test suite for new or updated code as appropriate
- [x] I've updated documentation (JSDoc, Markdown, etc.) for new or updated code as appropriate
- [x] I've highlighted breaking changes using the "BREAKING" category above as appropriate
- [x] I've prepared draft pull requests for clients and consumer packages to resolve any breaking changes

### CODE REVIEW

File: packages/profile-sync-controller/src/controllers/authentication/AuthenticationController.ts [hunk 1]
```diff
@@ -222,8 +222,6 @@ export default class AuthenticationController extends BaseController<
   }
 
   public performSignOut(): void {
-    this.#assertLoggedIn();
```
Comment (Prithpal-Sooriya) [inline packages/profile-sync-controller/src/controllers/authentication/AuthenticationController.ts:225]: We do not want to perform any assertions when signing out! We just need to reset/update state (see below state update)

Comment (Prithpal-Sooriya) [reply packages/profile-sync-controller/src/controllers/authentication/AuthenticationController.ts:225]: The intention for asserting the state was a good idea initially, but if there is some state mismatch, then they are unable to sign out!

**Review by mirceanis — APPROVED** [review]
Looks good

**Review by mathieuartu — APPROVED** [review]
LGTM!

**Pull request merged**